src/roiextractors/multiimagingextractor.py

```diff
@@ -53,8 +53,8 @@ def _check_consistency_between_imaging_extractors(self):
                 len(unique_values) == 1
             ), f"{property_message} is not consistent over the files (found {unique_values})."
 
-    def get_frames(self, frame_idxs: ArrayType, channel: int = 0) -> NumpyArray:
-        assert max(frame_idxs) < self._num_frames, "'frame_idxs' range beyond number of available frames!"
+    @check_get_frames_args
```

Comment:
@CodyCBakerPhD @h-mayorquin 
I was wondering if I could use this decorator here, but I noticed when a single frame is extracted while using this decorator, it only returns the first row of the frame:

https://github.com/catalystneuro/roiextractors/blob/e9080cc53daadb9685222429163dd00564506d9b/src/roiextractors/extraction_tools.py#L289-L290
What was the reasoning behind this? If this is intended, I can go back to do the checks without the decorator, just curious.

Reply:
I haven't been using the decorators for this personally - from past experience, they can produce very hard to debug issues. There's certainly a time and place for them, though, but I'm okay with not doing that decorators here.